src/dialog.rs

```diff
@@ -31,85 +29,14 @@ impl Dialog {
     }
 }
 
-pub trait DialogExt {
-    // -- manual --
+pub trait DialogExtManual {
     fn add_buttons(&self, buttons: &[(&str, i32)]);
-    // -- auto --
-    fn add_action_widget<T: IsA<Widget>>(&self, child: &T, response_id: i32);
-    fn add_button(&self, button_text: &str, response_id: i32) -> Widget;
-    fn get_action_area(&self) -> Widget;
-    fn get_content_area(&self) -> Box;
-    #[cfg(feature = "v3_12")]
-    fn get_header_bar(&self) -> Option<Widget>;
-    fn get_response_for_widget<T: IsA<Widget>>(&self, widget: &T) -> i32;
-    fn get_widget_for_response(&self, response_id: i32) -> Option<Widget>;
-    fn response(&self, response_id: i32);
-    fn run(&self) -> i32;
-    fn set_default_response(&self, response_id: i32);
-    fn set_response_sensitive(&self, response_id: i32, setting: bool);
 }
 
-impl<O: IsA<Dialog> + IsA<Object>> DialogExt for O {
+impl<O: DialogExt> DialogExtManual for O {
     fn add_buttons(&self, buttons: &[(&str, i32)]) {
         for &(text, id) in buttons {
-            Auto::add_button(self, text, id);
+            O::add_button(self, text, id);
```

Comment:
That's an interesting bug. Does `DialogExt` instead of `O` not work either? I guess there should be a 'FIXME' mentioning that `self.add_button` doesn't work on 1.8.